forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -482,28 +382,25 @@ pub async fn deploy_executables(
             "Saved",
             &format!("loader bytecode at {}", bin_path.display()),
         );
-        if let Some(loader_data_section_offset) =
-            loader_data_offset(&pkg.bytecode.bytes, &BlobId::default())?
-        {
-            if let ProgramABI::Fuel(mut fuel_abi) = pkg.program_abi.clone() {
-                println_action_green("Generating", "loader abi for the uploaded executable.");
-                let json_abi_path = out_dir.join(format!("{pkg_name}-loader-abi.json"));
-                let original_data_section = extract_data_offset(&pkg.bytecode.bytes).unwrap();
-                let offset_shift = original_data_section - loader_data_section_offset;
-                // if there are configurables in the abi we need to shift them by `offset_shift`.
-                let configurables = fuel_abi.configurables.clone().map(|configs| {
-                    configs
-                        .into_iter()
-                        .map(|config| Configurable {
-                            offset: config.offset - offset_shift as u64,
-                            ..config.clone()
-                        })
-                        .collect()
-                });
-                fuel_abi.configurables = configurables;
-                let json_string = serde_json::to_string_pretty(&fuel_abi)?;
-                std::fs::write(json_abi_path, json_string)?;
-            }
+        let loader_data_section_offset = loader.data_offset_in_code();
+        if let ProgramABI::Fuel(mut fuel_abi) = pkg.program_abi.clone() {
+            println_action_green("Generating", "loader abi for the uploaded executable.");
+            let json_abi_path = out_dir.join(format!("{pkg_name}-loader-abi.json"));
+            let original_data_section = extract_data_offset(&pkg.bytecode.bytes).unwrap();
+            let offset_shift = original_data_section - loader_data_section_offset;
+            // if there are configurables in the abi we need to shift them by `offset_shift`.
+            let configurables = fuel_abi.configurables.clone().map(|configs| {
+                configs
+                    .into_iter()
+                    .map(|config| Configurable {
+                        offset: config.offset - offset_shift as u64,
+                        ..config.clone()
+                    })
+                    .collect()
+            });
+            fuel_abi.configurables = configurables;
+            let json_string = serde_json::to_string_pretty(&fuel_abi)?;
+            std::fs::write(json_abi_path, json_string)?;
```

Comment:
Looks good to me, but I believe we need a test for this. Especially because we are not sure what is going to happen if the binary does not have a `data` section. In earlier implementation it was obvious that the function to calculate offset returns `None` in that case but now I am not sure what is going to happen. 

So if you are up for it, let's add a test which get's its oracle values (like the calculated shifted values with the previous implementation) and in a next commit let's apply this change and see if the calculated values are same.